File: guilib-api/src/main/java/net/developertobi/guilib/api/GuiApi.java
package net.developertobi.guilib.api;

import net.developertobi.guilib.api.gui.GuiHandler;
import net.developertobi.guilib.api.item.GuiItem;
import net.developertobi.guilib.api.item.GuiPos;
import net.developertobi.guilib.api.pagination.GuiPagination;
import net.kyori.adventure.text.Component;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.inventory.ItemStack;

import java.util.function.Consumer;

public interface GuiApi {

    GuiHandler getGuiHandler();

    GuiItem placeholder(Material type);
    GuiItem nextPage(ItemStack item, GuiPagination pagination);
    GuiItem previousPage(ItemStack item, GuiPagination pagination);
    GuiItem navigator(ItemStack item, String guiKey);
    GuiItem of(ItemStack item);
    GuiItem of(ItemStack item, GuiItemAction action);

    void openConfirmationGui(
        Player holder,
        Component title,
        ItemStack displayItem,
        Consumer<ItemStack> onAccept,
        Consumer<ItemStack> onDeny
    );
}

File: guilib-api/bin/main/net/developertobi/inventorylib/api/GuiApi.kt
package net.developertobi.inventorylib.api

import net.kyori.adventure.text.Component
import net.developertobi.inventorylib.api.inventory.GuiHandler
import net.developertobi.inventorylib.api.item.GuiItem
import net.developertobi.inventorylib.api.item.GuiPos
import net.developertobi.inventorylib.api.pagination.GuiPagination
import org.bukkit.Material
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

interface GuiApi {

    val inventoryHandler: GuiHandler

    fun placeholder(type: Material): GuiItem
    fun nextPage(item: ItemStack, pagination: GuiPagination): GuiItem
    fun previousPage(item: ItemStack, pagination: GuiPagination): GuiItem
    fun navigator(item: ItemStack, inventoryKey: String): GuiItem
    fun of(item: ItemStack): GuiItem
    fun of(item: ItemStack, action: ((GuiPos, GuiItem, InventoryClickEvent) -> Unit)): GuiItem

    fun openConfirmationInventory(
        holder: Player,
        title: Component,
        displayItem: ItemStack,
        onAccept: ((ItemStack) -> Unit),
        onDeny: ((ItemStack) -> Unit)
    )

}

